Get "Cannot run Project.afterEvaluate(Closure) when the project is already evaluated." when add some configuration

When I try to specify the checker framework version in the dependency block like:
```
dependencies {
   // ...
    val checkerFrameworkVersion = "3.25.0"
    compileOnly("org.checkerframework:checker-qual:$checkerFrameworkVersion")
    testCompileOnly("org.checkerframework:checker-qual:$checkerFrameworkVersion")
    implecmentation("org.checkerframework:checker-util:$checkerFrameworkVersion")
    checkerFramework("org.checkerframework:checker:$checkerFrameworkVersion")
   // ...
}
```
or add `skipVersionCheck = true` to the configuration:
```
configure<org.checkerframework.gradle.plugin.CheckerFrameworkExtension> {
    checkers = listOf(
        "org.checkerframework.checker.nullness.NullnessChecker"
    )
    excludeTests = true
    skipVersionCheck = true
}
```
I got the following errors:
```
Execution failed for task ':dependencies'.
> Failed to notify dependency resolution listener.
   > Cannot run Project.afterEvaluate(Closure) when the project is already evaluated.
   > Cannot run Project.afterEvaluate(Closure) when the project is already evaluated.
   > Cannot run Project.afterEvaluate(Closure) when the project is already evaluated.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
```

Can you help me to figure out why, please?
I use Gradle 7.4.2 with Kotlin build script and JDK 17.

I am seeing the same issue as @blackdiz.  Here is my branch:

https://github.com/msridhar/tainting-example/tree/called-methods-lombok

On that branch, here is what I see:
```
$ ./gradlew compileJava --console=plain --stacktrace
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :createCheckerFrameworkManifest
> Task :generateEffectiveLombokConfig FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateEffectiveLombokConfig'.
> Cannot run Project.afterEvaluate(Closure) when the project is already evaluated.

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':generateEffectiveLombokConfig'.
[...]
Caused by: org.gradle.api.InvalidUserCodeException: Cannot run Project.afterEvaluate(Closure) when the project is already evaluated.
	at org.gradle.api.internal.project.DefaultProject.failAfterProjectIsEvaluated(DefaultProject.java:1068)
	at org.gradle.api.internal.project.DefaultProject.afterEvaluate(DefaultProject.java:1062)
	at org.gradle.api.Project$afterEvaluate.call(Unknown Source)
	at org.checkerframework.gradle.plugin.CheckerFrameworkPlugin$1.execute(CheckerFrameworkPlugin.groovy:139)
	at org.checkerframework.gradle.plugin.CheckerFrameworkPlugin$1.execute(CheckerFrameworkPlugin.groovy)
	at org.gradle.api.internal.collections.CollectionFilter$1.execute(CollectionFilter.java:59)
	at org.gradle.api.internal.DefaultDomainObjectCollection.all(DefaultDomainObjectCollection.java:161)
	at org.gradle.api.internal.DefaultDomainObjectCollection.withType(DefaultDomainObjectCollection.java:203)
	at org.gradle.api.DomainObjectCollection$withType$0.call(Unknown Source)
	at org.checkerframework.gradle.plugin.CheckerFrameworkPlugin.handleLombokPlugin(CheckerFrameworkPlugin.groovy:105)
```

--- a/README.md
+++ b/README.md
@@ -12,7 +12,7 @@ Add the following to your `build.gradle` file:
 ```groovy
 plugins {
     // Checker Framework pluggable type-checking
-    id 'org.checkerframework' version '0.6.17'
+    id 'org.checkerframework' version '0.6.18'
 }
 
 apply plugin: 'org.checkerframework'
@@ -94,7 +94,7 @@ the definitions of the custom qualifiers.
 
 ### Specifying a Checker Framework version
 
-Version 0.6.17 of this plugin uses Checker Framework version 3.26.0 by default.
+Version 0.6.18 of this plugin uses Checker Framework version 3.26.0 by default.
 Anytime you upgrade to a newer version of this plugin,
 it might use a different version of the Checker Framework.
 
@@ -214,7 +214,7 @@ plugin to the top-level project. For example:
 
 ```groovy
 plugins {
-  id 'org.checkerframework' version '0.6.17' apply false
+  id 'org.checkerframework' version '0.6.18' apply false
 }
 
 subprojects { subproject ->
@@ -257,7 +257,7 @@ plugins {
   id "net.ltgt.errorprone" version "1.1.1" apply false
   // To do Checker Framework pluggable type-checking (and disable Error Prone), run:
   // ./gradlew compileJava -PuseCheckerFramework=true
-  id 'org.checkerframework' version '0.6.17' apply false
+  id 'org.checkerframework' version '0.6.18' apply false
 }
 
 if (!project.hasProperty("useCheckerFramework")) {

--- a/build.gradle
+++ b/build.gradle
@@ -38,7 +38,7 @@ dependencies {
 }
 
 group 'org.checkerframework'
-version '0.6.17'
+version '0.6.18'
 
 gradlePlugin {
     plugins {

--- a/src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy
+++ b/src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy
@@ -136,55 +136,52 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
           LOG.info(cmcLombokInteractionWarningMessage)
         }
 
-        project.afterEvaluate {
+        if (skipCheckerFramework(project, userConfig)) {
+          return
+        }
 
-          if (skipCheckerFramework(project, userConfig)) {
-            return
-          }
+        def delombokTasks = project.getTasks().findAll { task ->
+          task.name.contains("delombok")
+        }
 
-          def delombokTasks = project.getTasks().findAll { task ->
-            task.name.contains("delombok")
-          }
+        if (delombokTasks.size() != 0) {
 
-          if (delombokTasks.size() != 0) {
+          // change every compile task so that it:
+          // 1. depends on delombok, and
+          // 2. uses the delombok'd source code as its source set
 
-            // change every compile task so that it:
-            // 1. depends on delombok, and
-            // 2. uses the delombok'd source code as its source set
+          project.tasks.withType(AbstractCompile).all { compile ->
 
-            project.tasks.withType(AbstractCompile).all { compile ->
+            // find the right delombok task
+            def delombokTask = delombokTasks.find { task ->
 
-              // find the right delombok task
-              def delombokTask = delombokTasks.find { task ->
+              if (task.name.endsWith("delombok")) {
+                // special-case the main compile task because it's just named "compileJava"
+                // without anything else
+                compile.name.equals("compileJava")
+              } else {
+                // "delombok" is 8 characters.
+                compile.name.contains(task.name.substring(8))
+              }
+            }
 
-                if (task.name.endsWith("delombok")) {
-                  // special-case the main compile task because its just named "compileJava"
-                  // without anything else
-                  compile.name.equals("compileJava")
-                } else {
-                  // "delombok" is 8 characters.
-                  compile.name.contains(task.name.substring(8))
-                }
+            // delombokTask can still be null; for example, if the code contains a compileScala task
+            // Also, if we're skipping test tasks, don't bother delombok'ing them.
+            if (delombokTask != null && !compile.getSource().isEmpty()
+                    && !(userConfig.excludeTests && delombokTask.name.toLowerCase().contains("test"))) {
+
+              // The lombok plugin's default formatting is pretty-printing, without the @Generated annotations
+              // that we need to recognize lombok'd code.
+              delombokTask.format.put('generated', 'generate')
+
+              if (userConfig.suppressLombokWarnings) {
+                // Also re-add suppress warnings annotations so that we don't get warnings from generated
+                // code.
+                delombokTask.format.put('suppressWarnings', 'generate')
               }
 
-              // delombokTask can still be null; for example, if the code contains a compileScala task
-              // Also, if we're skipping test tasks, don't bother delombok'ing them.
-              if (delombokTask != null && !compile.getSource().isEmpty()
-                      && !(userConfig.excludeTests && delombokTask.name.toLowerCase().contains("test"))) {
-
-                // The lombok plugin's default formatting is pretty-printing, without the @Generated annotations
-                // that we need to recognize lombok'd code.
-                delombokTask.format.put('generated', 'generate')
-
-                if (userConfig.suppressLombokWarnings) {
-                  // Also re-add suppress warnings annotations so that we don't get warnings from generated
-                  // code.
-                  delombokTask.format.put('suppressWarnings', 'generate')
-                }
-
-                compile.dependsOn(delombokTask)
-                compile.setSource(delombokTask.target.getAsFile().get())
-              }
+              compile.setSource(delombokTask.target.getAsFile().get())
+              compile.dependsOn(delombokTask)
             }
           }
         }
@@ -247,7 +244,6 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
           }
         }
 
-        handleLombokPlugin(project, userConfig)
         // Only attempt to add each dependency once.
         project.getGradle().removeListener(this)
       }
@@ -256,6 +252,7 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
       void afterResolve(ResolvableDependencies resolvableDependencies) {}
     })
 
+    handleLombokPlugin(project, userConfig)
     configureTasks(project, AbstractCompile, { AbstractCompile compile ->
       def ext = compile.extensions.findByName("checkerFramework")
       if (ext == null) {

